R/class-bal_accuracy.R

```diff
@@ -81,6 +81,13 @@ bal_accuracy_vec <- function(truth,
                              case_weights = NULL,
                              event_level = yardstick_event_level(),
                              ...) {
+  if (is_class_pred(truth)) {
```

Comment:
From [R4DS](https://r4ds.had.co.nz/functions.html#when-should-you-write-a-function)

> ... the “do not repeat yourself” (or DRY) principle. The more repetition you have in your code, the more places you need to remember to update when things change (and they always do!), and the more likely you are to create bugs over time.

Reply:
Is there a constructor or some other commonly used function to put this in? We'll have to remember to add this to every classification metric that uses hard class predictions.

Maybe we should add one that is automatically invoked somewhere?

Reply:
> Is there a constructor or some other commonly used function to put this in? We'll have to remember to add this to every classification metric that uses hard class predictions.

Short answer: no.

at first I wanted to put this in the metric summarizers such as `class_metric_summarizer()`, to avoid copying all this code around. However, when I did that, all the `*_vec()` functions didn't work because they don't call the metric summarizers. 

From https://github.com/tidymodels/yardstick/issues/341 we have that `finalize_estimator()` is the typical culprit in errors, and since that is the first thing called in the `*_vec()` functions I sadly had to put that code there.

This is of cause predicated on the assumption that we want both `accuracy()` and `accuracy_vec()` to be able to deal with `class_pred` which seems reasonable.

Reply:
ok., I would at least make it a function that gets invoked a lot.

Reply:
done here [tidymodels/yardstick@`90fa6af` (#422)](https://github.com/tidymodels/yardstick/pull/422/commits/90fa6af29bb5aa3995dd2a5c4bca49991a838241)